tests/alphabet.cc

```diff
@@ -38,25 +38,21 @@ TEST_CASE("Mata::IntAlphabet") {
 }
 
 TEST_CASE("Mata::OnTheFlyAlphabet::add_symbols_from()") {
-    OnTheFlyAlphabet alphabet{};
-    StringToSymbolMap symbol_map{ { "a", 4 }, { "b", 2 }, { "c", 10 } };
-    alphabet.add_symbols_from(symbol_map);
-
-    auto symbols{alphabet.get_alphabet_symbols() };
+    OnTheFlyAlphabet alphabet{ { "a", 4 }, { "b", 2 }, { "c", 10 } };
+    auto symbols{ alphabet.get_alphabet_symbols() };
     Mata::Util::OrdVector<Symbol> expected{ 4, 2, 10 };
     CHECK(symbols == expected);
     CHECK(alphabet.get_next_value() == 11);
-    CHECK(alphabet.get_symbol_map() == symbol_map);
+    CHECK(alphabet.get_symbol_map() == OnTheFlyAlphabet::StringToSymbolMap{ { "a", 4 }, { "b", 2 }, { "c", 10 } });
```

Comment:
So we cannot get rid of the StringToSymbolMap completely?

Reply:
`StringAlphabet` (here `OnTheFlyAlphabet`) is using `StringToSymbolMap` (`std::unordered_map<std::string, Mata::Symbol>`) only as an internal member variable type for storing the mapping. One will use `StringToSymbolMap` only if they want to work directly with the internal members of `OnTheFlyAlphabet`. Alternatively, we can remove the internal type declaration entirely and use `std::unordered_map<std::string, Mata::Symbol>` everywhere. `StringAlphabet::get_symbol_map()` will then return this underlying type as well.

Reply:
Does it have to be accessible like this through the interface? Maybe not?

Reply:
Depends. If it is not accessible, we have to use `std::unordered_map<std::string, Mata::Symbol>` in the interface. If that is what we want, I can replace it. It is rather a lengthy type. I think that the use of `StringToSymbolMap` (or `StringSymbolMap`?) is warranted in this case of `StringAlphabet`.

Reply:
The question is whether we need to use this type in the interface at all, not whether we nee to rename it.

Reply:
We will remove the accessor entirely. StringToSymobolMap will be removed, too. All operations will be done on alphabet objects, such as equivalence etc.

Reply:
I made a note about resolving the alphabet-related suggestions in the PR for alphabet refactorization.